src/error.rs

```diff
@@ -0,0 +1,30 @@
+//! Errors!
+
+use miette::Diagnostic;
+use thiserror::Error;
+
+/// Gotta love a newtyped Result
+pub type Result<T> = std::result::Result<T, AxoprocessError>;
+
+/// An error from executing a Command
+#[derive(Debug, Error, Diagnostic)]
+pub enum AxoprocessError {
+    /// The command fundamentally failed to execute (usually means it didn't exist)
+    #[error("failed to {summary}")]
+    Exec {
+        /// Summary of what the Command was trying to do
+        summary: String,
+        /// What failed
+        #[source]
+        cause: std::io::Error,
+    },
```

Comment:
we could embed the entire Command and print it as context or something?

Reply:
Yeah, I like that idea.